src/libstd/sys/unix/thread.rs

```diff
@@ -88,28 +79,16 @@ pub mod guard {
     #[cfg(any(target_os = "linux", target_os = "android"))]
     unsafe fn get_stack_start() -> *mut libc::c_void {
         let mut attr: libc::pthread_attr_t = mem::zeroed();
-        if pthread_getattr_np(pthread_self(), &mut attr) != 0 {
-            panic!("failed to get thread attributes");
-        }
+        eq!(pthread_getattr_np(pthread_self(), &mut attr), 0);
```

Comment:
> This reduces clarity of the error and makes harder to find the cause IMO.

Could you expand on this some more? I expect this panic message to be more informative as it includes the line number, expression, and return value if it's not zero.

> Also, if debug_assert_eq!, does what I think it does, this code behave incorrectly and most likely will SIGSEGV/SIGILL

That's a good point, I'll switch back to `assert_eq!`

> I would defer this change until somebody gets rid of pthread usage here.

I don't quite follow, do you think we *can* get rid of this usage?